### PR TITLE
Text to speech work

### PR DESCRIPTION
**Pull request by nmarshak1337**
Adds text-to-speech functionality from MeSpeak.js to the sound driver. The text-to-speech feature in Mars-Game (https://github.com/virtual-world-framework/mars-game/pull/480) depends on the changes here.

### CODE REVIEW

File: support/client/lib/index.html [hunk 1]
```diff
@@ -46,6 +46,7 @@
 
     <script type="text/javascript" src="alea.js"></script>
     <script type="text/javascript" src="mash.js"></script>
+    <script type="text/javascript" src="mespeak.js"></script>
```
Comment (davideaster) [inline support/client/lib/index.html]: Can this move to the sound driver? The loader shouldn't be dependent on any particular driver.

The [cesium driver] (https://github.com/virtual-world-framework/vwf/blob/master/support/client/lib/vwf/model/cesium.js) has an example of loading the Cesium library this way:

    define( [ "module", "vwf/model", "vwf/utility", "vwf/utility/color", "vwf/model/cesium/Cesium" ],
        function( module, model, utility, Color, Cesium ) { ... } );

File: support/client/lib/vwf/model/sound.js [hunk 2]
```diff
@@ -283,6 +287,10 @@ define( [ "module", "vwf/model" ], function( module, model ) {
                 this.playOnLoad = soundDefinition.playOnLoad;
             }
 
+            if (this.soundDefinition.useTextToSpeech !== undefined) {
```
Comment (AmbientOSX) [inline support/client/lib/vwf/model/sound.js]: It is a VWF convention to space things as such: if ( this.soundDefinition.useTextToSpeech !== undefined )

This is a comment that could be applied throughout but I'll save the spam and just say to look out for it :)

File: support/client/lib/vwf/model/sound.js [hunk 3]
```diff
@@ -251,6 +254,7 @@ define( [ "module", "vwf/model" ], function( module, model ) {
         allowMultiplay: false,
         soundDefinition: null,
         playOnLoad: false,
+        useTextToSpeech: false,
```
Comment (AmbientOSX) [inline support/client/lib/vwf/model/sound.js]: May want to define some TTS default parameters here?

**Comment by AmbientOSX** [comment]
Looks pretty good... When I tried playing the voices seemed a bit quick so I think we'd want to slow them down a bit but I guess that's more something for @kadst43 to weigh in on. :)

**Comment by nmarshak1337** [comment]
Closing because meSpeak is GPL. We may revisit at a later date.

**Pull request closed without merging**